main.js

```diff
@@ -0,0 +1,182 @@
+//variables
+let totalExpenses = 0;
+let expenses = [];
+
+//DOM elements
+let form = document.querySelector(".input-form");
+let expensesList = document.querySelector(".expenses");
+let TexpensesContainer = document.querySelector(".Texpenses");
+let clearHistoryBtn = document.querySelector(".clear-history-btn");
+let BalanceContainer = document.querySelector(".balance");
+
+// ======= Program Initialization ======= 
+// Prompt user for total income and start the application
+
+let value = prompt("enter your total income to start : ");
+let Tincome = sanitizer(value);
+
+function runApp(){
+
+loadStoredExpenses();
+
+updateFinance();
+
+form.addEventListener("submit" ,(event)=> {
+  event.preventDefault();
+  addToList();
+});
+
+clearHistoryBtn.addEventListener("click" , ()=>{
+  localStorage.removeItem("expenses");
+  expenses = [];
+  totalExpenses = 0;
+  renderArray(expenses);
+  updateFinance();
+})
+
+};
+
+runApp();
+
+// ======= End of Initialization ======= 
+
+
+// ======= Functions ======= 
+
+function sanitizer(value){
+
+  if(value === null){
+    alert("You pressed Cancel. Exiting app");
+    window.close();
+    return;
+  }
+
+  let data = Number(value);
+  while(!Number.isInteger(data) || value.trim() === ""){
+    value = prompt("enter a valid integer : ");
+    if(value === null){
+      alert("You pressed Cancel. Exiting app");
+      window.close();
+      return;
+    }
+    data = Number(value);
+  }
+
+  return data;
+}
+
+function loadStoredExpenses(){
+  expenses = JSON.parse(localStorage.getItem("expenses")) || [];
+  totalExpenses = expenses.reduce((sum, item) => sum + item.cost, 0);
+  renderArray(expenses);
+}
+
+function updateFinance(){ 
+  updateTincome();
+  updateTexpenses();
+  updateBalance();
+  }
+  
+  function updateTincome(){
+    let totalRemainedContainer = document.querySelector(".totalRemained");
+    totalRemained = Tincome - totalExpenses;
+    totalRemainedContainer.innerHTML = `Total : ${totalRemained}`;
+  }
+  
+  function updateTexpenses(){
+    let TexpensesContainer = document.querySelector(".Texpenses");
+    TexpensesContainer.innerHTML = `Total expenses : ${totalExpenses}`;
+  }
+  
+  function updateBalance(){
+    let balanceContainer = document.querySelector(".balance");
+    
+    let balance = totalRemained;  
+    
+    balanceContainer.innerHTML = `Balance: ${balance}`;
+    
+    if(balance < Tincome * 0.2){   // If less than 20% of total income
```

Comment:
If you mean `totalIncome` by using `Tincome` it is better to write it completely.
Also, starting a word with a capital letter in JS has some conventions and use cases (e.g., class), so it affects your code's readability.